frontend/src/concepts/pipelines/content/createRun/useRunFormData.ts

```diff
@@ -55,6 +61,37 @@ const useUpdateData = <T extends PipelineCoreResourceKF>(
   }, [setData]);
 };
 
+const useUpdatePipelineRun = (
+  setFunction: UpdateObjectAtPropAndValue<RunFormData>,
+  initialData?: PipelineRunKF | PipelineRunJobKF,
+) => {
+  const { getJobInformation } = usePipelinesAPI();
+  const { data: pipelineRunJob, loading } = getJobInformation(initialData);
+
+  const updatedSetFunction = React.useCallback<UpdateObjectAtPropAndValue<RunFormData>>(
+    (key, pipeline) => {
+      if (!loading && isPipeline(pipeline)) {
+        setFunction(key, pipeline);
+        setFunction('params', getParams(pipeline));
+        const nameDesc: ValueOf<RunFormData> = {
+          name: initialData?.name ? `Duplicate of ${initialData.name}` : '',
+          description: pipelineRunJob?.description ?? initialData?.description ?? '',
+        };
+        setFunction('nameDesc', nameDesc);
```

Comment:
I'm still working on this -- we should re-use existing details and only apply our description if somehow the resource doesn't have one. We should not touch the name -- that can be picked up from the earlier flows.

Ashique and I talked about this -- I wasn't able to work out the TS logic in time. We can swing back.